src/api.js

```diff
@@ -280,6 +280,55 @@ api.declare({
 });
 
 api.declare({
+  name: 'badges',
+  title: 'Latest Build Status Badge',
+  description: [
+    'Checks the status of the latest build of a given branch ',
+    'and returns corresponding badge image.',
+  ].join('\n'),
+  stability: 'experimental',
+  method: 'get',
+  route: '/status/:owner/:repo/:branch',
+}, async function(req, res) {
+  // Extract owner, repo and branch from request into variables
+  let {owner, repo, branch} = req.params;
+
+  // Look up the installation ID in Azure. If no such owner in the table, no error thrown
+  let ownerInfo = await this.OwnersDirectory.load({owner}, true);
+
+  if (ownerInfo) {
+    try {
+      let instGithub = await this.github.getInstallationGithub(ownerInfo.installationId);
+      // Statuses is an array of status objects
+      let statuses = await instGithub.repos.getStatuses({owner, repo, ref: branch});
+      let status = statuses.find(statusObject => statusObject.context.startsWith(`${this.cfg.app.statusContext}`));
+
+      // The weird switch is here for the future changes
+      switch (status.state) {
+        case 'success':
+          res.writeHead(200, {'Content-Type': 'image/png'});
+          res.reply({status: 'success'});
+          break;
+        case 'failure':
+          res.writeHead(200, {'Content-Type': 'image/png'});
+          res.sendFile('../assets/failure.png');
+          break;
```

Comment:
I'm not sure how to do that :) But don't give me hints just yet, I'd like to try and think of something